symbiflow.py

```diff
@@ -0,0 +1,232 @@
+import os
+import subprocess
+import time
+import collections
+import json
+import re
+import shutil
+import sys
+import glob
+import datetime
+import asciitable
+import edalize
+
+from toolchain import Toolchain
+from utils import Timed
+
+class VPR(Toolchain):
+    '''VPR using Yosys for synthesis'''
+    carries = (False,)
+
+    def __init__(self, rootdir):
+        Toolchain.__init__(self, rootdir)
+        self.toolchain = 'vpr'
+        self.files = []
+
+    def run(self):
+        with Timed(self, 'bit-all'):
+            shutil.rmtree(self.out_dir, ignore_errors=True)
+            os.makedirs(self.out_dir, exist_ok = True)
+
+            for f in self.srcs:
+                self.files.append({'name' : os.path.realpath(f), 'file_type': 'verilogSource'})
+
+            if self.pcf:
+                print("Adding pcf file", self.pcf)
+                self.files.append({'name' : os.path.realpath(self.pcf), 'file_type': 'SDC'})
+
+            chip = self.family + self.device
+
+            self.edam = {
+                'files' : self.files,
+                'name' : self.project_name,
+                'toplevel' : self.top,
+                'tool_options' : {'symbiflow' :
```

Comment:
Wrapping of this dictionary is a bit weird.

Reply:
fixed